File: Whatsapp/OpenWA/sdk/php/src/Resources/HealthResource.php
<?php

declare(strict_types=1);

namespace OpenWA\Resources;

use OpenWA\Http\HttpExecutor;

/**
 * Health resource — connectivity and readiness probes.
 *
 * Backed by src/modules/health/health.controller.ts.
 */
class HealthResource
{
    private HttpExecutor $http;

    public function __construct(HttpExecutor $http)
    {
        $this->http = $http;
    }

    /** @return array<string,mixed> */
    public function check(): array
    {
        return $this->http->request('GET', '/api/health') ?? [];
    }

    /** @return array<string,mixed> */
    public function live(): array
    {
        return $this->http->request('GET', '/api/health/live') ?? [];
    }

    /** @return array<string,mixed> */
    public function ready(): array
    {
        return $this->http->request('GET', '/api/health/ready') ?? [];
    }
}

File: OpenWA/sdk/java/src/main/java/com/rmyndharis/openwa/resources/HealthResource.java
package com.rmyndharis.openwa.resources;

import com.rmyndharis.openwa.OpenWAClient;
import com.rmyndharis.openwa.http.HttpMethod;
import com.rmyndharis.openwa.model.HealthReadyResponse;
import com.rmyndharis.openwa.model.HealthResponse;

/** Health resource — connectivity and readiness probes. */
public final class HealthResource {
    private final OpenWAClient client;

    public HealthResource(OpenWAClient client) {
        this.client = client;
    }

    /** General health (also returns the running version). */
    public HealthResponse check() {
        return client.request(HttpMethod.GET, "/api/health", null, null, HealthResponse.class);
    }

    /** Kubernetes liveness probe. */
    public HealthResponse live() {
        return client.request(HttpMethod.GET, "/api/health/live", null, null, HealthResponse.class);
    }

    /** Kubernetes readiness probe — checks both DB connections. */
    public HealthReadyResponse ready() {
        return client.request(HttpMethod.GET, "/api/health/ready", null, null, HealthReadyResponse.class);
    }
}
